### PR TITLE
refinement typing go brrrrrr

### PR DESCRIPTION
**Pull request by BoxyUwU**
r? @oli-obk

### CODE REVIEW

**Review by oli-obk — CHANGES_REQUESTED** [review]
Probably a good idea to squash some (most?) commits

File: compiler/rustc_mir_transform/src/refinements.rs [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+struct LocalRefines(Box<[Option<RefineRange>]>);
+
+impl LocalRefines {
+    /// Sets every refinement to `..` / `_`
+    fn clear_all(&mut self) {
+        for refine in self.0.iter_mut() {
+            *refine = Some(RefineRange::from(0..=(u128::MAX)));
+        }
```
Comment (LingMan) [inline compiler/rustc_mir_transform/src/refinements.rs:218]:
```suggestion
        self.0.fill(Some(RefineRange::from(0..=u128::MAX)));
```

**Comment by apiraino** [comment]
my understanding is that this is now:

@rustbot label -S-waiting-on-review +S-waiting-on-author

**Comment by JohnCSimon** [comment]
Ping from triage: 
@BoxyUwU - Can you please post the status of this PR?  Thank you.

**Comment by BoxyUwU** [comment]
just gonna close this because im tired and dont want to think about this PR lol

**Pull request closed without merging**